File: algorithms/implementation/p5_sherlock_and_squares/sherlock_and_squares.py
#!/bin/python3

import sys
from math import ceil, floor, sqrt

t = int(input().strip())
for a0 in range(t):
    temp = input().split(" ")
    beginCeil = int(ceil(sqrt(int(temp[0]))))
    endFloor = int(floor(sqrt(int(temp[1]))))
    print(endFloor - beginCeil + 1)

File: algorithms/implementation/p5_sherlock_and_squares/sherlock_and_squares.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(int a0 = 0; a0 < t; a0++){
            int begin = in.nextInt();
            int end = in.nextInt();
            int beginCeil = (int)Math.ceil(Math.sqrt(begin));
            int endFloor = (int)Math.floor(Math.sqrt(end));
            System.out.println(endFloor - beginCeil + 1);
        }
    }
}
